client-core/src/hd_seed.rs

```diff
@@ -39,13 +39,14 @@ impl HDSeed {
     pub fn derive_key_pair(
         &self,
         network: Network,
-        address_type: AddressType,
+        address_type: Option<AddressType>,
         index: u32,
     ) -> Result<(PublicKey, PrivateKey)> {
         let coin_type = get_bip44_coin_type_from_network(network);
         let account = match address_type {
-            AddressType::Transfer => 0,
-            AddressType::Staking => 1,
+            Some(AddressType::Transfer) => 0,
+            Some(AddressType::Staking) => 1,
+            None => 2,
```

Comment:
I don't have enough context for this, but "multisig" address is a transfer address, so why this separation? Just create a transfer address and get an associated public key?

Reply:
Transfer addresses in HD wallet are 1-of-n transfer addresses which are created using `wallet_client.new_transfer_address()`.